Duplicati/Library/Backend/Telegram/EncryptedFileSessionStore.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using TLSharp.Core;
+
+namespace Duplicati.Library.Backend
+{
+    public class EncryptedFileSessionStore : ISessionStore
+    {
+        private static readonly object m_lockObj = new object();
+        private static readonly uint[] m_lookup32 = CreateLookup32();
+
+        private readonly string m_teleDataPath;
+        private readonly string m_password;
+        private readonly SHA256 m_sha = SHA256.Create();
+        private static readonly ConcurrentDictionary<string, byte[]> m_userIdLastSessionMap = new ConcurrentDictionary<string, byte[]>();
+
+        public EncryptedFileSessionStore(string password)
+        {
+            if (string.IsNullOrWhiteSpace(password))
+            {
+                throw new ArgumentNullException(nameof(password));
+            }
+
+            var appData = Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData);
+            m_teleDataPath = Path.Combine(appData, nameof(Duplicati), nameof(Telegram));
```

Comment:
We allow for [rebranding](https://github.com/duplicati/duplicati/wiki/Branding-and-OEM-customization).  Instead of `nameof(Duplicati)`, would it be better to reference [AutoUpdateSettings.AppName](https://github.com/duplicati/duplicati/blob/v2.0.5.111-2.0.5.111_canary_2020-09-26/Duplicati/Library/AutoUpdater/AutoUpdateSettings.cs#L160)?